propose.md

```diff
@@ -0,0 +1,80 @@
+# New proposal for rich-text type
+
+## Problem
+The previous representation of the rich-text type had already reached its limits. The idea was to separate the content (Words) from the formatting (Selections). It was defined as follow:
+
+    Words = {
+        word: string,
+        left: [selection], # the selections starting at this word
+        right: [selection], # the selections ending at this word
+
+    }
+
+    Selections = {
+        left: word,
+        leftPos: number,
+        right :word,
+        rightPos: number,
+        style: object
+    }
+
+    RichText = {
+        words: [Words],
+        selections: [Selections]
+    }
+
+The main problem is that it's quite hard to determine the behaviour of the selection if a word gets updated from another peer. Let's see another problem to really see where the issue lies.
+
+If we have the following text: "Longsword *is* a *kind* of sword" and two peers, A and B.
+At a certain moment, these actions are triggered:
+- A: set " a " to bold, resulting in the merge of three selections, A now has
+  "Longsword *is a kind* of sword"
+- B:  unset bold on "is", B now has
+  "Longsword is a *kind* of sword"
+
+Now if these modifications are sent, what should happen? A has lost the memory of the selection on *kind* and B has no memory of *is*. So we could imagine to store all the previous modifications, but what happend when theire's 100's of concurrent users creating and deleting selections? The list of past selections grows bigger and bigger.
+
+## Proposed solution
+On the other side, Yjs objects.that are deleted aren't deleted immediately. They are destroyed by a garbage collector when it is known for sure that no peer can make modifications prior to the deletion of the object. Using this, it is possible to imagine an ordered list of actions, very similar to operational transforms (https://github.com/ottypes/docs) that result in the creation of selections (formatting). The list is garbage collected so it doesn't grow indefinitely. To prevent the issue of relative adressing within word, strings are stored as a list of characters, and selections are pointing at the object holding each character.
+
+The structure of the text-type would then be:
+
+    Char = {
+        val: string,
+        left: [selection],
+        right: [selection],
```

Comment:
Imagine the word "abcd". Since a.right === b.left, I'm pretty sure, that it is possible, to do this with only one property.

It could always point to the left of the character. So in order to select a only, you'd set `a.selectedBy = sel` and  `b.selectedBy = set`. If the cursor is to the left of a, then only `a.selectedBy = sel`. But `sel.start = a`; and `sel.end = a`.

Just as a general rule: define it as small as possible. This will also reduce the number of cases you have to handle for several upcoming issues